src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -54,13 +54,9 @@ public override void MoveFile(string sourceFullPath, string destFullPath)
         {
             // The desired behavior for Move(source, dest) is to not overwrite the destination file
             // if it exists. Since rename(source, dest) will replace the file at 'dest' if it exists,
-            // a check is added before rename is called. Note that this is a race condition. Possible
-            // better solutions would be to use link/unlink or platform specific functions instead of
-            // rename(source, dest). Regardless of the implementation, an IOException should be thrown if 
-            // 'dest' refers to a file that already exists.
-            if (File.Exists(destFullPath))
-                throw new IOException(SR.GetResourceString(SR.IO_AlreadyExists_Name, destFullPath));
-            while (Interop.libc.rename(sourceFullPath, destFullPath) < 0)
+            // link/unlink are used instead. Note that the act of unlinking will trigger a deletion
+            // notice for the FileSystemWatcher
+            while (Interop.libc.link(sourceFullPath, destFullPath) < 0)
```

Comment:
It is related to #2537, I was leaving it in place until a decision was reached on how File.Move should handle an invalid source path, but it can be removed.